Sources/Sentry/SentryTracer.m

```diff
@@ -116,7 +116,10 @@ - (void)captureTransaction
     transaction.transaction = self.name;
     [_hub captureEvent:transaction withScope:_hub.scope];
 
-    [_hub.scope clearSpan:self];
+    [_hub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+        if (span == self)
+            [self->_hub.scope setSpan:nil];
```

Comment:
Always using `{}` is advised to avoid issues

```suggestion
        if (span == self) {
            [self->_hub.scope setSpan:nil];
        }
```